CodeConverter/Util/FromRoslyn/ITypeSymbolExtensions.cs

```diff
@@ -254,6 +254,36 @@ public static bool IsUnexpressibleTypeParameterConstraint(this ITypeSymbol typeS
             return false;
         }
 
+        public static bool IsFloatingNumericType(this ITypeSymbol? type)
+        {
+            if (type != null) {
+                switch (type.SpecialType) {
+                    case SpecialType.System_Single:
+                    case SpecialType.System_Double:
+                    case SpecialType.System_Decimal:
+                        return true;
+                }
+            }
+
+            return false;
+        }
+
+        public static bool IsIntegerNumericType(this ITypeSymbol? type)
+        {
+            if (type != null) {
+                switch (type.SpecialType) {
+                    case SpecialType.System_Int16:
```

Comment:
Shouldn't this also include
```csharp
                     case SpecialType.System_Byte:
                    case SpecialType.System_SByte:
```

I'm not sure about char, the fact it isn't in the existing method below means it probably isn't required, but might be worth checking the behaviour.

Reply:
If byte and sbyte are needed, then consider implementing some of these methods in terms of the others to avoid inconsistencies in future changes.
e.g. `IsNumericType(type) && !IsFloatingNumericType(type)`